internal/app/init/pkg/system/services/trustd.go

```diff
@@ -76,3 +80,26 @@ func (t *Trustd) Runner(data *userdata.UserData) (runner.Runner, error) {
 		restart.WithType(restart.Forever),
 	), nil
 }
+
+// HealthFunc implements the HealthcheckedService interface
+func (t *Trustd) HealthFunc(*userdata.UserData) health.Check {
+	return func(ctx context.Context) error {
+		var d net.Dialer
+		conn, err := d.DialContext(ctx, "tcp", fmt.Sprintf("%s:%d", "localhost", constants.TrustdPort))
+		if err != nil {
+			return err
+		}
+
+		return conn.Close()
+	}
+}
+
+// HealthSettings implements the HealthcheckedService interface
+func (t *Trustd) HealthSettings(*userdata.UserData) *health.Settings {
+	return &health.DefaultSettings
+}
+
+// Verify healthchecked interface
+var (
+	_ system.HealthcheckedService = &Trustd{}
```

Comment:
Remind me again with the blank import does?

Reply:
this is not a blank import, this a bit ugly, but I don't know a better way.

healthchecks were implemented as additional interface, `HealthcheckedService` so that we don't have to add dummy healthcheck functions to the services which naturally don't support them. But the problem is that if `HealtcheckedService` definition changes, and service isn't updated, code will compile, but health checks won't run. so this line simply tests that type `Trustd` implements interface `HealthcheckedService`, and code won't compile if it doesn't.

So variable `_` is a Go convention for anonymous variable (kind of /dev/null). We just assign value of the type to the interface, and type conversion would be successful only if type implements the interface

Reply:
Ahh right. I have gotten around this other ways but I think this makes sense in this case.